### DISCUSSION TITLE
Where is the client secret?

### DISCUSSION THREAD

**Discussion by Ali229** [llamafilm/tesla-http-proxy-addon#130] [Q&A]
The addon is requiring client id and secret, I have the ID but i cannot for the life of find the secret?
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/11164440/2e6eb7a9-bf39-4e09-aa81-6400b2f2f84b)

**Comment by llamafilm** [comment]
You need to change the grant type to include “machine to machine”

**Comment by Ali229** [comment]
Got it thanks, needed to answer no to open-source during tesla developer setup
